deepchecks/nlp/text_data.py

```diff
@@ -497,6 +497,15 @@ def is_sampled(self, n_samples: t.Optional[int]):
         if n_samples is None:
             return False
         return self.n_samples > n_samples
+    @classmethod
+    def from_hugging_face(cls, obj:t.Any) -> 'TextData':
+
+        if set(['tokens', 'ner_tags']).issubset(obj.column_names):
```

Comment:
will work on it this week

Reply:
@noamzbr. What is the purpose of this method if we are just using all the TextData arguments?

Reply:
Hi @dsjoench, the purpose is to create a TextData from a huggingFace Dataset directly. Doing that though you still need control over the configurable parameters of a TextData object, thus the need for arguments. If HF Datasets had these parameters internally (task type, metadata cols etc) we could read it directly from them but this is not that case.

Reply:
I added those configurable parameters of a TextData object. 

This is another dataset example that uses text and label to create a text_classification task.   Let me know if you want me to add it as another test case
dataset = load_dataset('imdb')
textData = TextData.from_hugging_face(dataset, tokenized_text=dataset['text'],
                                                      label=dataset['label'], task_type='text_classification')

Reply:
Hi @dsjoench, sorry if I have previously been unclear - I didn't mean to use the columns directly (as you've shown in your example, this makes the actual dataset object pretty useless), but instead to add an *optional* way to change the default names of the fields used from the huggingface dataset, while still enabling the lesser arguments such as task_type (mandatory), name, embeddings, metadata columns etc

Reply:
so if I understand correctly, instead of us checking whether specific columns exist in HF dataset to choose a task_type, we want the client to choose whatever task_type they want and optionally provide columns to create a TextData. What if those default names of fields from HF dataset were not provided? Do we look for fields and set it for client?

Reply:
Yes for the first part, and about the default names - I'm actually not sure. What names you encounter most for text and labels? Is there any common name or is it just different for each dataset?

Reply:
it's different for each dataset. I've come across a dataset that has only 'tokens', 'ner_tags' and a dataset that has only 'text', 'label'. I haven't come across anything that have all of above, so it makes it easy to know  dataset with 'tokens', 'ner_tags' belongs to token_classification and vice versa.